Angus/1518 pixel highlight

veggiesaurus opened the pull request:
Closes #1518 

Also fixes the blurry lines on the colorbar hover line by clamping the line position to mid-pixel

veggiesaurus committed:
adds pixel highlighting when hovering over the colorbar

veggiesaurus committed:
adjust mouse handling

veggiesaurus committed:
delay highlighting by 500 ms

YuHsuan-Hwang commented:
looks great :thumbsup:
In the image view setting, "Cursor Info" is for showing/hiding the text and the bar. We can change it to "Interactive" (default true) and include enabling/disabling pixel highlight

veggiesaurus committed:
Merge branch 'dev' into angus/1518_pixel_highlight

veggiesaurus committed:
Only show highlights when colorbar is set to interactive

veggiesaurus committed:
refactoring: rename showHoverInfo -> interactive

veggiesaurus commented:
> In the image view setting, "Cursor Info" is for showing/hiding the text and the bar. We can change it to "Interactive" (default true) and include enabling/disabling pixel highlight

@YuHsuan-Hwang Ok, I've implemented that. I've also refactored the `showHoverInfo` setting to `interactive` and the `showHoverInfo` observable in the component itself to `isHovering`

kswang1029 reviewed (approved):
this works nicely 👍 

Later on we can see how we can freeze the colorbar cursor so that users can make a png for presentation for example.
Probably we can consider to use click to enable/disable cursor freeze.
